crates/sui-quorum-driver/src/lib.rs

```diff
@@ -0,0 +1,250 @@
+        quorum_driver: &Arc<QuorumDriver<A>>,
+        certificate: CertifiedTransaction,
+        timeout: Duration,
+    ) -> SuiResult<(CertifiedTransaction, TransactionEffects)> {
+        let effects = quorum_driver
+            .validators
+            .load()
+            .process_certificate(certificate.clone(), timeout)
+            .instrument(tracing::debug_span!("process_cert"))
+            .await?;
+        Ok((certificate, effects))
+    }
+}
+
+impl<A> QuorumDriverHandler<A>
+where
+    A: AuthorityAPI + Send + Sync + 'static + Clone,
+{
+    pub async fn execute_transaction(
+        &self,
+        request: ExecuteTransactionRequest,
+    ) -> SuiResult<ExecuteTransactionResponse> {
+        let ExecuteTransactionRequest {
+            transaction,
+            request_type,
+        } = request;
+        match request_type {
+            ExecuteTransactionRequestType::ImmediateReturn => {
+                self.task_sender
+                    .lock()
+                    .await
+                    .send(QuorumTask::ProcessTransaction(transaction))
+                    .await
+                    .map_err(|err| SuiError::QuorumDriverCommunicationError {
+                        error: err.to_string(),
+                    })?;
+                Ok(ExecuteTransactionResponse::ImmediateReturn)
+            }
+            ExecuteTransactionRequestType::WaitForTxCert(timeout) => {
+                let certificate = QuorumDriverHandler::process_transaction(
+                    &self.quorum_driver,
+                    transaction,
+                    timeout,
+                )
+                .instrument(tracing::debug_span!("process_tx"))
+                .await?;
+                self.task_sender
+                    .lock()
+                    .await
+                    .send(QuorumTask::ProcessCertificate(certificate.clone()))
+                    .await
+                    .map_err(|err| SuiError::QuorumDriverCommunicationError {
+                        error: err.to_string(),
+                    })?;
+                Ok(ExecuteTransactionResponse::TxCert(Box::new(certificate)))
+            }
+            ExecuteTransactionRequestType::WaitForEffectsCert(timeout) => {
+                let instant = Instant::now();
+                let certificate = QuorumDriverHandler::process_transaction(
+                    &self.quorum_driver,
+                    transaction,
+                    timeout,
+                )
+                .instrument(tracing::debug_span!("process_tx"))
+                .await?;
+                let response = QuorumDriverHandler::process_certificate(
+                    &self.quorum_driver,
+                    certificate,
+                    timeout - instant.elapsed(),
+                )
+                .instrument(tracing::debug_span!("process_cert"))
+                .await?;
+                Ok(ExecuteTransactionResponse::EffectsCert(Box::new(response)))
+            }
```

Comment:
Evan built a very similar pattern for sending commands over a channel for lock_service.rs, I wonder if its worth abstracting that mechanism so we don't keep re-inventing it.